pkg/controller/vsphere/util.go

```diff
@@ -0,0 +1,92 @@
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+
+	"gopkg.in/gcfg.v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	osconfig "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	globalInfrastuctureName  = "cluster"
+	openshiftConfigNamespace = "openshift-config"
+)
+
+// vSphereConfig is a copy of the Kubernetes vSphere cloud provider config type
+// that contains the fields we need.  Unfortunately, we can't easily import
+// either the legacy or newer cloud provider code here, so we're just
+// duplicating part of the type and parsing it ourselves using the same gcfg
+// library for now.
+type vSphereConfig struct {
+	Labels struct {
+		Zone   string `gcfg:"zone"`
+		Region string `gcfg:"region"`
+	}
+}
+
+func getInfrastructure(cfg *rest.Config) (*configv1.Infrastructure, error) {
+	if cfg == nil {
+		return nil, errors.New("No REST config")
+	}
+
+	clientset, err := osconfig.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	infrav1 := clientset.ConfigV1().Infrastructures()
+
+	return infrav1.Get(globalInfrastuctureName, metav1.GetOptions{})
+}
+
+func getVSphereConfig(cfg *rest.Config) (*vSphereConfig, error) {
+	if cfg == nil {
+		return nil, errors.New("No REST config")
```

Comment:
nit: use lowercase for passing through error

Reply:
Fixed. This bit is gone.